test/functional/test_runner.py

```diff
@@ -269,9 +269,20 @@ def main():
     test_list = []
     if tests:
         # Individual tests have been specified. Run specified tests that exist
-        # in the ALL_SCRIPTS list. Accept the name with or without .py extension.
+        # in the ALL_SCRIPTS list. Accept names with or without a .py extension.
+        # Specified tests can contain paths and wildcards, but the paths must be
+        # coherent, e.g. the same path as that provided to call test_runner.py.
+        # Examples:
+        #   `test/functional/test_runner.py test/functional/wallet*`
+        #   `test/functional/test_runner.py ./test/functional/wallet*`
+        #   `test_runner.py wallet*`
+        #   but not:
+        #   `test/functional/test_runner.py wallet*`
+        # Multiple wildcards can be passed:
+        #   `test_runner.py tool* mempool*`
         tests = [test + ".py" if ".py" not in test else test for test in tests]
         for test in tests:
+            test = test.split("/")[-1]
```

Comment:
I liked the simplicity of the one-line diff but made 2 minor additional changes:
- moved `tests = [test + ".py" if ".py" not in test else test for test in tests]` into the main loop to avoid iterating twice through `tests`
- added a local variable `script` to avoid mutating the `test` iterator

Happy to revert to the one-line version if that is preferred.